add-on/panels/test/panel_test.js

```diff
@@ -350,50 +350,94 @@ describe("loop.panel", function() {
 
           sinon.assert.calledOnce(requestStubs.LoginToFxA);
         });
-      });
 
-      it("should show a signout entry when user is authenticated", function() {
-        loop.storedRequests.GetUserProfile = { email: "test@example.com" };
+        it("should close the menu on clicking sign in", function() {
+          var view = mountTestComponent();
+
+          TestUtils.Simulate.click(view.getDOMNode()
+                                     .querySelector(".entry-settings-signin"));
 
-        mountTestComponent();
+          expect(view.state.showMenu).eql(false);
+        });
+
+        it("should close the panel on clicking sign in", function() {
+          var view = mountTestComponent();
+
+          TestUtils.Simulate.click(view.getDOMNode()
+                                     .querySelector(".entry-settings-signin"));
 
-        sinon.assert.calledWithExactly(document.mozL10n.get,
-                                       "settings_menu_item_signout");
-        sinon.assert.neverCalledWith(document.mozL10n.get,
-                                     "settings_menu_item_signin");
+          sinon.assert.calledOnce(fakeWindow.close);
+        });
       });
 
-      it("should show an account entry when user is authenticated", function() {
-        LoopMochaUtils.stubLoopRequest({
-          GetUserProfile: function() { return { email: "test@example.com" }; }
+      describe("UserLoggedIn", function() {
+        it("should show a signout entry when user is authenticated", function() {
+          loop.storedRequests.GetUserProfile = { email: "test@example.com" };
+
+          mountTestComponent();
+
+          sinon.assert.calledWithExactly(document.mozL10n.get,
+              "settings_menu_item_signout");
+          sinon.assert.neverCalledWith(document.mozL10n.get,
```

Comment:
This isn't your fault as its existing code, but can we improve these tests whilst we're touching them? Using a test for a function call when testing if a button is displayed or not isn't ideal.

Can you drop the two sinon.assert calls and replace them with:

```
var view = mountTestComponent();

expect(view.getDOMNode().querySelectorAll(".entry-settings-signout"))
  .to.have.length.of(1);
expect(view.getDOMNode().querySelectorAll(".entry-settings-signin"))
  .to.have.length.of(0);
```